direct/src/directnotify/RotatingLog.py

```diff
@@ -28,33 +36,34 @@ def __init__(self, path="./log_file", hourInterval=24, megabyteLimit=1024):
         if megabyteLimit is not None:
             self.sizeLimit = megabyteLimit*1024*1024
 
-    def __del__(self):
+    def __del__(self) -> None:
         self.close()
 
-    def close(self):
+    def close(self) -> None:
         if hasattr(self, "file"):
             self.file.flush()
             self.file.close()
             self.closed = self.file.closed
             del self.file
         else:
-            self.closed = 1
+            self.closed = True
 
-    def shouldRotate(self):
+    def shouldRotate(self) -> bool:
         """
         Returns a bool about whether a new log file should
         be created and written to (while at the same time
         stopping output to the old log file and closing it).
         """
         if not hasattr(self, "file"):
-            return 1
+            return True
         if self.timeLimit is not None and time.time() > self.timeLimit:
-            return 1
+            return True
         if self.sizeLimit is not None and self.file.tell() > self.sizeLimit:
-            return 1
-        return 0
+            return True
+        return False
 
-    def filePath(self):
+    def filePath(self) -> str:
+        assert self.sizeLimit is not None
```

Comment:
You've identified an oversight in the code - the assert is improper because this case ought to be handled.  I'll check in a commit.

Reply:
Done in c63a92bdec8043ec883ae8afc57ecf1670325d70 - you can remove the assert.